// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ActivityTypes model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes
 * @version 9.4.5
 */
var ActivityTypes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityTypes</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes
   */
  function ActivityTypes() {
    _classCallCheck(this, ActivityTypes);

    ActivityTypes.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ActivityTypes, [{
    key: "getType",
    value:
    /**
         * Returns If the author type is `user`, only `conversation:read` is supported.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes.TypeEnum}
         */
    function getType() {
      return this.type;
    }
    /**
     * Sets If the author type is `user`, only `conversation:read` is supported.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes.TypeEnum} type If the author type is `user`, only `conversation:read` is supported.
     */

  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActivityTypes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes} The populated <code>ActivityTypes</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityTypes();

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ActivityTypes;
}();
/**
 * If the author type is `user`, only `conversation:read` is supported.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityTypes.TypeEnum} type
 */


ActivityTypes.prototype['type'] = undefined;
/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */

ActivityTypes['TypeEnum'] = {
  /**
   * value: "conversation:read"
   * @const
   */
  "conversation:read": "conversation:read",

  /**
   * value: "typing:start"
   * @const
   */
  "typing:start": "typing:start",

  /**
   * value: "typing:stop"
   * @const
   */
  "typing:stop": "typing:stop"
};
var _default = ActivityTypes;
exports["default"] = _default;
