// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ActivityAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AuthorWebhook = _interopRequireDefault(require("./AuthorWebhook"));

var _SourceWebhook = _interopRequireDefault(require("./SourceWebhook"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ActivityAllOf model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ActivityAllOf
 * @version 9.4.5
 */
var ActivityAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ActivityAllOf</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityAllOf
   */
  function ActivityAllOf() {
    _classCallCheck(this, ActivityAllOf);

    ActivityAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ActivityAllOf, [{
    key: "getSource",
    value:
    /**
         * Returns The source of the activity.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook}
         */
    function getSource() {
      return this.source;
    }
    /**
     * Sets The source of the activity.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source The source of the activity.
     */

  }, {
    key: "setSource",
    value: function setSource(source) {
      this['source'] = source;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook}
         */

  }, {
    key: "getAuthor",
    value: function getAuthor() {
      return this.author;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook} author
     */

  }, {
    key: "setAuthor",
    value: function setAuthor(author) {
      this['author'] = author;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ActivityAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityAllOf} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ActivityAllOf} The populated <code>ActivityAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ActivityAllOf();

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], _SourceWebhook["default"]);
        }

        if (data.hasOwnProperty('author')) {
          obj['author'] = _AuthorWebhook["default"].constructFromObject(data['author']);
        }
      }

      return obj;
    }
  }]);

  return ActivityAllOf;
}();
/**
 * The source of the activity.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SourceWebhook} source
 */


ActivityAllOf.prototype['source'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/AuthorWebhook} author
 */

ActivityAllOf.prototype['author'] = undefined;
var _default = ActivityAllOf;
exports["default"] = _default;
