// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AppListResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _App = _interopRequireDefault(require("./App"));

var _Links = _interopRequireDefault(require("./Links"));

var _Meta = _interopRequireDefault(require("./Meta"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AppListResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AppListResponse
 * @version 9.4.5
 */
var AppListResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AppListResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AppListResponse
   */
  function AppListResponse() {
    _classCallCheck(this, AppListResponse);

    AppListResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AppListResponse, [{
    key: "getApps",
    value:
    /**
         * Returns List of returned apps.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/App>}
         */
    function getApps() {
      return this.apps;
    }
    /**
     * Sets List of returned apps.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/App>} apps List of returned apps.
     */

  }, {
    key: "setApps",
    value: function setApps(apps) {
      this['apps'] = apps;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta}
         */

  }, {
    key: "getMeta",
    value: function getMeta() {
      return this.meta;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta} meta
     */

  }, {
    key: "setMeta",
    value: function setMeta(meta) {
      this['meta'] = meta;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Links}
         */

  }, {
    key: "getLinks",
    value: function getLinks() {
      return this.links;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} links
     */

  }, {
    key: "setLinks",
    value: function setLinks(links) {
      this['links'] = links;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AppListResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppListResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AppListResponse} The populated <code>AppListResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AppListResponse();

        if (data.hasOwnProperty('apps')) {
          obj['apps'] = _ApiClient["default"].convertToType(data['apps'], [_App["default"]]);
        }

        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _Meta["default"].constructFromObject(data['meta']);
        }

        if (data.hasOwnProperty('links')) {
          obj['links'] = _Links["default"].constructFromObject(data['links']);
        }
      }

      return obj;
    }
  }]);

  return AppListResponse;
}();
/**
 * List of returned apps.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/App>} apps
 */


AppListResponse.prototype['apps'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta} meta
 */

AppListResponse.prototype['meta'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} links
 */

AppListResponse.prototype['links'] = undefined;
var _default = AppListResponse;
exports["default"] = _default;
