// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AppResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _App = _interopRequireDefault(require("./App"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AppResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AppResponse
 * @version 9.4.5
 */
var AppResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AppResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AppResponse
   */
  function AppResponse() {
    _classCallCheck(this, AppResponse);

    AppResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AppResponse, [{
    key: "getApp",
    value:
    /**
         * Returns The app.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/App}
         */
    function getApp() {
      return this.app;
    }
    /**
     * Sets The app.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/App} app The app.
     */

  }, {
    key: "setApp",
    value: function setApp(app) {
      this['app'] = app;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AppResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AppResponse} The populated <code>AppResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AppResponse();

        if (data.hasOwnProperty('app')) {
          obj['app'] = _ApiClient["default"].convertToType(data['app'], _App["default"]);
        }
      }

      return obj;
    }
  }]);

  return AppResponse;
}();
/**
 * The app.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/App} app
 */


AppResponse.prototype['app'] = undefined;
var _default = AppResponse;
exports["default"] = _default;
