// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AppSubSchema.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AppSubSchema model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AppSubSchema
 * @version 9.4.5
 */
var AppSubSchema = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AppSubSchema</code>.
   * The app that triggered the events.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AppSubSchema
   */
  function AppSubSchema() {
    _classCallCheck(this, AppSubSchema);

    AppSubSchema.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AppSubSchema, [{
    key: "getId",
    value:
    /**
         * Returns The unique ID of the app.
         * @return {String}
         */
    function getId() {
      return this.id;
    }
    /**
     * Sets The unique ID of the app.
     * @param {String} id The unique ID of the app.
     */

  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AppSubSchema</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AppSubSchema} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AppSubSchema} The populated <code>AppSubSchema</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AppSubSchema();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return AppSubSchema;
}();
/**
 * The unique ID of the app.
 * @member {String} id
 */


AppSubSchema.prototype['id'] = undefined;
var _default = AppSubSchema;
exports["default"] = _default;
