// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AttachmentUploadBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AttachmentUploadBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AttachmentUploadBody
 * @version 9.4.5
 */
var AttachmentUploadBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AttachmentUploadBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentUploadBody
   * @param source {File} 
   */
  function AttachmentUploadBody(source) {
    _classCallCheck(this, AttachmentUploadBody);

    AttachmentUploadBody.initialize(this, source);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AttachmentUploadBody, [{
    key: "getSource",
    value:
    /**
         * @return {File}
         */
    function getSource() {
      return this.source;
    }
    /**
     * @param {File} source
     */

  }, {
    key: "setSource",
    value: function setSource(source) {
      this['source'] = source;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, source) {
      obj['source'] = source;
    }
    /**
     * Constructs a <code>AttachmentUploadBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentUploadBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentUploadBody} The populated <code>AttachmentUploadBody</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AttachmentUploadBody();

        if (data.hasOwnProperty('source')) {
          obj['source'] = _ApiClient["default"].convertToType(data['source'], File);
        }
      }

      return obj;
    }
  }]);

  return AttachmentUploadBody;
}();
/**
 * @member {File} source
 */


AttachmentUploadBody.prototype['source'] = undefined;
var _default = AttachmentUploadBody;
exports["default"] = _default;
