// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AttachmentMediaTokenBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AttachmentMediaTokenBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AttachmentMediaTokenBody
 * @version 9.4.5
 */
var AttachmentMediaTokenBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AttachmentMediaTokenBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentMediaTokenBody
   * @param paths {Array.<String>} An array of strings representing the list of attachment paths used to generate the media JWT. One token will be generated for each path. Each token will be valid for 2 hours by default. This value can be modified, see [App Settings](#operation/createApp) for more information.
   */
  function AttachmentMediaTokenBody(paths) {
    _classCallCheck(this, AttachmentMediaTokenBody);

    AttachmentMediaTokenBody.initialize(this, paths);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AttachmentMediaTokenBody, [{
    key: "getPaths",
    value:
    /**
         * Returns An array of strings representing the list of attachment paths used to generate the media JWT. One token will be generated for each path. Each token will be valid for 2 hours by default. This value can be modified, see [App Settings](#operation/createApp) for more information.
         * @return {Array.<String>}
         */
    function getPaths() {
      return this.paths;
    }
    /**
     * Sets An array of strings representing the list of attachment paths used to generate the media JWT. One token will be generated for each path. Each token will be valid for 2 hours by default. This value can be modified, see [App Settings](#operation/createApp) for more information.
     * @param {Array.<String>} paths An array of strings representing the list of attachment paths used to generate the media JWT. One token will be generated for each path. Each token will be valid for 2 hours by default. This value can be modified, see [App Settings](#operation/createApp) for more information.
     */

  }, {
    key: "setPaths",
    value: function setPaths(paths) {
      this['paths'] = paths;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, paths) {
      obj['paths'] = paths;
    }
    /**
     * Constructs a <code>AttachmentMediaTokenBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentMediaTokenBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentMediaTokenBody} The populated <code>AttachmentMediaTokenBody</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AttachmentMediaTokenBody();

        if (data.hasOwnProperty('paths')) {
          obj['paths'] = _ApiClient["default"].convertToType(data['paths'], ['String']);
        }
      }

      return obj;
    }
  }]);

  return AttachmentMediaTokenBody;
}();
/**
 * An array of strings representing the list of attachment paths used to generate the media JWT. One token will be generated for each path. Each token will be valid for 2 hours by default. This value can be modified, see [App Settings](#operation/createApp) for more information.
 * @member {Array.<String>} paths
 */


AttachmentMediaTokenBody.prototype['paths'] = undefined;
var _default = AttachmentMediaTokenBody;
exports["default"] = _default;
