// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AttachmentResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _AttachmentSchema = _interopRequireDefault(require("./AttachmentSchema"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AttachmentResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AttachmentResponse
 * @version 9.4.5
 */
var AttachmentResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AttachmentResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentResponse
   */
  function AttachmentResponse() {
    _classCallCheck(this, AttachmentResponse);

    AttachmentResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AttachmentResponse, [{
    key: "getAttachment",
    value:
    /**
         * Returns The uploaded attachment object.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentSchema}
         */
    function getAttachment() {
      return this.attachment;
    }
    /**
     * Sets The uploaded attachment object.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentSchema} attachment The uploaded attachment object.
     */

  }, {
    key: "setAttachment",
    value: function setAttachment(attachment) {
      this['attachment'] = attachment;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>AttachmentResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentResponse} The populated <code>AttachmentResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AttachmentResponse();

        if (data.hasOwnProperty('attachment')) {
          obj['attachment'] = _ApiClient["default"].convertToType(data['attachment'], _AttachmentSchema["default"]);
        }
      }

      return obj;
    }
  }]);

  return AttachmentResponse;
}();
/**
 * The uploaded attachment object.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentSchema} attachment
 */


AttachmentResponse.prototype['attachment'] = undefined;
var _default = AttachmentResponse;
exports["default"] = _default;
