// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/AttachmentDeleteBody.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The AttachmentDeleteBody model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/AttachmentDeleteBody
 * @version 9.4.5
 */
var AttachmentDeleteBody = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AttachmentDeleteBody</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentDeleteBody
   * @param mediaUrl {String} The media URL used for a file or image message.
   */
  function AttachmentDeleteBody(mediaUrl) {
    _classCallCheck(this, AttachmentDeleteBody);

    AttachmentDeleteBody.initialize(this, mediaUrl);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(AttachmentDeleteBody, [{
    key: "getMediaUrl",
    value:
    /**
         * Returns The media URL used for a file or image message.
         * @return {String}
         */
    function getMediaUrl() {
      return this.mediaUrl;
    }
    /**
     * Sets The media URL used for a file or image message.
     * @param {String} mediaUrl The media URL used for a file or image message.
     */

  }, {
    key: "setMediaUrl",
    value: function setMediaUrl(mediaUrl) {
      this['mediaUrl'] = mediaUrl;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, mediaUrl) {
      obj['mediaUrl'] = mediaUrl;
    }
    /**
     * Constructs a <code>AttachmentDeleteBody</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentDeleteBody} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/AttachmentDeleteBody} The populated <code>AttachmentDeleteBody</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AttachmentDeleteBody();

        if (data.hasOwnProperty('mediaUrl')) {
          obj['mediaUrl'] = _ApiClient["default"].convertToType(data['mediaUrl'], 'String');
        }
      }

      return obj;
    }
  }]);

  return AttachmentDeleteBody;
}();
/**
 * The media URL used for a file or image message.
 * @member {String} mediaUrl
 */


AttachmentDeleteBody.prototype['mediaUrl'] = undefined;
var _default = AttachmentDeleteBody;
exports["default"] = _default;
