// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ClientAssociation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ClientType = _interopRequireDefault(require("./ClientType"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClientAssociation model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ClientAssociation
 * @version 9.4.5
 */
var ClientAssociation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClientAssociation</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAssociation
   */
  function ClientAssociation() {
    _classCallCheck(this, ClientAssociation);

    ClientAssociation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClientAssociation, [{
    key: "getType",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientType}
         */
    function getType() {
      return this.type;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientType} type
     */

  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns The id of the client being referenced.
         * @return {String}
         */

  }, {
    key: "getClientId",
    value: function getClientId() {
      return this.clientId;
    }
    /**
     * Sets The id of the client being referenced.
     * @param {String} clientId The id of the client being referenced.
     */

  }, {
    key: "setClientId",
    value: function setClientId(clientId) {
      this['clientId'] = clientId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClientAssociation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAssociation} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientAssociation} The populated <code>ClientAssociation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClientAssociation();

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ClientType["default"].constructFromObject(data['type']);
        }

        if (data.hasOwnProperty('clientId')) {
          obj['clientId'] = _ApiClient["default"].convertToType(data['clientId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ClientAssociation;
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientType} type
 */


ClientAssociation.prototype['type'] = undefined;
/**
 * The id of the client being referenced.
 * @member {String} clientId
 */

ClientAssociation.prototype['clientId'] = undefined;
var _default = ClientAssociation;
exports["default"] = _default;
