// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ClientResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Client = _interopRequireDefault(require("./Client"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ClientResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ClientResponse
 * @version 9.4.5
 */
var ClientResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ClientResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ClientResponse
   */
  function ClientResponse() {
    _classCallCheck(this, ClientResponse);

    ClientResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ClientResponse, [{
    key: "getClient",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Client}
         */
    function getClient() {
      return this.client;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Client} client
     */

  }, {
    key: "setClient",
    value: function setClient(client) {
      this['client'] = client;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ClientResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ClientResponse} The populated <code>ClientResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ClientResponse();

        if (data.hasOwnProperty('client')) {
          obj['client'] = _Client["default"].constructFromObject(data['client']);
        }
      }

      return obj;
    }
  }]);

  return ClientResponse;
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Client} client
 */


ClientResponse.prototype['client'] = undefined;
var _default = ClientResponse;
exports["default"] = _default;
