// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationJoinEventAllOfPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ConversationTruncated = _interopRequireDefault(require("./ConversationTruncated"));

var _UserTruncated = _interopRequireDefault(require("./UserTruncated"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConversationJoinEventAllOfPayload model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationJoinEventAllOfPayload
 * @version 9.4.5
 */
var ConversationJoinEventAllOfPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationJoinEventAllOfPayload</code>.
   * The payload of the event. The contents of this object depend on the type of event.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationJoinEventAllOfPayload
   */
  function ConversationJoinEventAllOfPayload() {
    _classCallCheck(this, ConversationJoinEventAllOfPayload);

    ConversationJoinEventAllOfPayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConversationJoinEventAllOfPayload, [{
    key: "getConversation",
    value:
    /**
         * Returns The conversation in which the user was added.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated}
         */
    function getConversation() {
      return this.conversation;
    }
    /**
     * Sets The conversation in which the user was added.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} conversation The conversation in which the user was added.
     */

  }, {
    key: "setConversation",
    value: function setConversation(conversation) {
      this['conversation'] = conversation;
    }
    /**
         * Returns The user that joined the conversation.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated}
         */

  }, {
    key: "getUser",
    value: function getUser() {
      return this.user;
    }
    /**
     * Sets The user that joined the conversation.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated} user The user that joined the conversation.
     */

  }, {
    key: "setUser",
    value: function setUser(user) {
      this['user'] = user;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConversationJoinEventAllOfPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationJoinEventAllOfPayload} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationJoinEventAllOfPayload} The populated <code>ConversationJoinEventAllOfPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationJoinEventAllOfPayload();

        if (data.hasOwnProperty('conversation')) {
          obj['conversation'] = _ApiClient["default"].convertToType(data['conversation'], _ConversationTruncated["default"]);
        }

        if (data.hasOwnProperty('user')) {
          obj['user'] = _ApiClient["default"].convertToType(data['user'], _UserTruncated["default"]);
        }
      }

      return obj;
    }
  }]);

  return ConversationJoinEventAllOfPayload;
}();
/**
 * The conversation in which the user was added.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} conversation
 */


ConversationJoinEventAllOfPayload.prototype['conversation'] = undefined;
/**
 * The user that joined the conversation.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated} user
 */

ConversationJoinEventAllOfPayload.prototype['user'] = undefined;
var _default = ConversationJoinEventAllOfPayload;
exports["default"] = _default;
