// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationListResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Conversation = _interopRequireDefault(require("./Conversation"));

var _Links = _interopRequireDefault(require("./Links"));

var _Meta = _interopRequireDefault(require("./Meta"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConversationListResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationListResponse
 * @version 9.4.5
 */
var ConversationListResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationListResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationListResponse
   */
  function ConversationListResponse() {
    _classCallCheck(this, ConversationListResponse);

    ConversationListResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConversationListResponse, [{
    key: "getConversations",
    value:
    /**
         * Returns List of returned conversations.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Conversation>}
         */
    function getConversations() {
      return this.conversations;
    }
    /**
     * Sets List of returned conversations.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Conversation>} conversations List of returned conversations.
     */

  }, {
    key: "setConversations",
    value: function setConversations(conversations) {
      this['conversations'] = conversations;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta}
         */

  }, {
    key: "getMeta",
    value: function getMeta() {
      return this.meta;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta} meta
     */

  }, {
    key: "setMeta",
    value: function setMeta(meta) {
      this['meta'] = meta;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Links}
         */

  }, {
    key: "getLinks",
    value: function getLinks() {
      return this.links;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} links
     */

  }, {
    key: "setLinks",
    value: function setLinks(links) {
      this['links'] = links;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConversationListResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationListResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationListResponse} The populated <code>ConversationListResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationListResponse();

        if (data.hasOwnProperty('conversations')) {
          obj['conversations'] = _ApiClient["default"].convertToType(data['conversations'], [_Conversation["default"]]);
        }

        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _Meta["default"].constructFromObject(data['meta']);
        }

        if (data.hasOwnProperty('links')) {
          obj['links'] = _Links["default"].constructFromObject(data['links']);
        }
      }

      return obj;
    }
  }]);

  return ConversationListResponse;
}();
/**
 * List of returned conversations.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Conversation>} conversations
 */


ConversationListResponse.prototype['conversations'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta} meta
 */

ConversationListResponse.prototype['meta'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} links
 */

ConversationListResponse.prototype['links'] = undefined;
var _default = ConversationListResponse;
exports["default"] = _default;
