// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationListFilter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConversationListFilter model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationListFilter
 * @version 9.4.5
 */
var ConversationListFilter = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationListFilter</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationListFilter
   */
  function ConversationListFilter() {
    _classCallCheck(this, ConversationListFilter);

    ConversationListFilter.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConversationListFilter, [{
    key: "getUserId",
    value:
    /**
         * Returns The user's id. One of `userId` or `userExternalId` is required, but not both.
         * @return {String}
         */
    function getUserId() {
      return this.userId;
    }
    /**
     * Sets The user's id. One of `userId` or `userExternalId` is required, but not both.
     * @param {String} userId The user's id. One of `userId` or `userExternalId` is required, but not both.
     */

  }, {
    key: "setUserId",
    value: function setUserId(userId) {
      this['userId'] = userId;
    }
    /**
         * Returns The external Id of the user. One of `userId` or `userExternalId` is required, but not both.
         * @return {String}
         */

  }, {
    key: "getUserExternalId",
    value: function getUserExternalId() {
      return this.userExternalId;
    }
    /**
     * Sets The external Id of the user. One of `userId` or `userExternalId` is required, but not both.
     * @param {String} userExternalId The external Id of the user. One of `userId` or `userExternalId` is required, but not both.
     */

  }, {
    key: "setUserExternalId",
    value: function setUserExternalId(userExternalId) {
      this['userExternalId'] = userExternalId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConversationListFilter</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationListFilter} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationListFilter} The populated <code>ConversationListFilter</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationListFilter();

        if (data.hasOwnProperty('userId')) {
          obj['userId'] = _ApiClient["default"].convertToType(data['userId'], 'String');
        }

        if (data.hasOwnProperty('userExternalId')) {
          obj['userExternalId'] = _ApiClient["default"].convertToType(data['userExternalId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ConversationListFilter;
}();
/**
 * The user's id. One of `userId` or `userExternalId` is required, but not both.
 * @member {String} userId
 */


ConversationListFilter.prototype['userId'] = undefined;
/**
 * The external Id of the user. One of `userId` or `userExternalId` is required, but not both.
 * @member {String} userExternalId
 */

ConversationListFilter.prototype['userExternalId'] = undefined;
var _default = ConversationListFilter;
exports["default"] = _default;
