// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadDestination.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConversationMessageDeliveryPayloadDestination model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadDestination
 * @version 9.4.5
 */
var ConversationMessageDeliveryPayloadDestination = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationMessageDeliveryPayloadDestination</code>.
   * A nested object representing the destination of the message.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadDestination
   */
  function ConversationMessageDeliveryPayloadDestination() {
    _classCallCheck(this, ConversationMessageDeliveryPayloadDestination);

    ConversationMessageDeliveryPayloadDestination.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConversationMessageDeliveryPayloadDestination, [{
    key: "getType",
    value:
    /**
         * Returns An identifier for the channel to which a message was sent to. May include one of \"web\", \"ios\", \"android\", \"messenger\", \"viber\", \"telegram\", \"wechat\", \"line\", \"twilio\", \"api\", \"notification\", or any other channel.
         * @return {String}
         */
    function getType() {
      return this.type;
    }
    /**
     * Sets An identifier for the channel to which a message was sent to. May include one of \"web\", \"ios\", \"android\", \"messenger\", \"viber\", \"telegram\", \"wechat\", \"line\", \"twilio\", \"api\", \"notification\", or any other channel.
     * @param {String} type An identifier for the channel to which a message was sent to. May include one of \"web\", \"ios\", \"android\", \"messenger\", \"viber\", \"telegram\", \"wechat\", \"line\", \"twilio\", \"api\", \"notification\", or any other channel.
     */

  }, {
    key: "setType",
    value: function setType(type) {
      this['type'] = type;
    }
    /**
         * Returns Identifier indicating which integration the message was sent to.
         * @return {String}
         */

  }, {
    key: "getIntegrationId",
    value: function getIntegrationId() {
      return this.integrationId;
    }
    /**
     * Sets Identifier indicating which integration the message was sent to.
     * @param {String} integrationId Identifier indicating which integration the message was sent to.
     */

  }, {
    key: "setIntegrationId",
    value: function setIntegrationId(integrationId) {
      this['integrationId'] = integrationId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConversationMessageDeliveryPayloadDestination</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadDestination} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadDestination} The populated <code>ConversationMessageDeliveryPayloadDestination</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationMessageDeliveryPayloadDestination();

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }

        if (data.hasOwnProperty('integrationId')) {
          obj['integrationId'] = _ApiClient["default"].convertToType(data['integrationId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ConversationMessageDeliveryPayloadDestination;
}();
/**
 * An identifier for the channel to which a message was sent to. May include one of \"web\", \"ios\", \"android\", \"messenger\", \"viber\", \"telegram\", \"wechat\", \"line\", \"twilio\", \"api\", \"notification\", or any other channel.
 * @member {String} type
 */


ConversationMessageDeliveryPayloadDestination.prototype['type'] = undefined;
/**
 * Identifier indicating which integration the message was sent to.
 * @member {String} integrationId
 */

ConversationMessageDeliveryPayloadDestination.prototype['integrationId'] = undefined;
var _default = ConversationMessageDeliveryPayloadDestination;
exports["default"] = _default;
