// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationReadEventAllOfPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Activity = _interopRequireDefault(require("./Activity"));

var _ConversationTruncated = _interopRequireDefault(require("./ConversationTruncated"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConversationReadEventAllOfPayload model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationReadEventAllOfPayload
 * @version 9.4.5
 */
var ConversationReadEventAllOfPayload = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationReadEventAllOfPayload</code>.
   * The payload of the event. The contents of this object depend on the type of event.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationReadEventAllOfPayload
   */
  function ConversationReadEventAllOfPayload() {
    _classCallCheck(this, ConversationReadEventAllOfPayload);

    ConversationReadEventAllOfPayload.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConversationReadEventAllOfPayload, [{
    key: "getConversation",
    value:
    /**
         * Returns The conversation in which the message was read.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated}
         */
    function getConversation() {
      return this.conversation;
    }
    /**
     * Sets The conversation in which the message was read.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} conversation The conversation in which the message was read.
     */

  }, {
    key: "setConversation",
    value: function setConversation(conversation) {
      this['conversation'] = conversation;
    }
    /**
         * Returns The activity that was sent.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Activity}
         */

  }, {
    key: "getActivity",
    value: function getActivity() {
      return this.activity;
    }
    /**
     * Sets The activity that was sent.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Activity} activity The activity that was sent.
     */

  }, {
    key: "setActivity",
    value: function setActivity(activity) {
      this['activity'] = activity;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConversationReadEventAllOfPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationReadEventAllOfPayload} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationReadEventAllOfPayload} The populated <code>ConversationReadEventAllOfPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationReadEventAllOfPayload();

        if (data.hasOwnProperty('conversation')) {
          obj['conversation'] = _ApiClient["default"].convertToType(data['conversation'], _ConversationTruncated["default"]);
        }

        if (data.hasOwnProperty('activity')) {
          obj['activity'] = _ApiClient["default"].convertToType(data['activity'], _Activity["default"]);
        }
      }

      return obj;
    }
  }]);

  return ConversationReadEventAllOfPayload;
}();
/**
 * The conversation in which the message was read.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationTruncated} conversation
 */


ConversationReadEventAllOfPayload.prototype['conversation'] = undefined;
/**
 * The activity that was sent.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Activity} activity
 */

ConversationReadEventAllOfPayload.prototype['activity'] = undefined;
var _default = ConversationReadEventAllOfPayload;
exports["default"] = _default;
