// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadMessage.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConversationMessageDeliveryPayloadMessage model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadMessage
 * @version 9.4.5
 */
var ConversationMessageDeliveryPayloadMessage = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationMessageDeliveryPayloadMessage</code>.
   * The message that was sent.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadMessage
   */
  function ConversationMessageDeliveryPayloadMessage() {
    _classCallCheck(this, ConversationMessageDeliveryPayloadMessage);

    ConversationMessageDeliveryPayloadMessage.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConversationMessageDeliveryPayloadMessage, [{
    key: "getId",
    value:
    /**
         * Returns A string representing the ID of the message.
         * @return {String}
         */
    function getId() {
      return this.id;
    }
    /**
     * Sets A string representing the ID of the message.
     * @param {String} id A string representing the ID of the message.
     */

  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConversationMessageDeliveryPayloadMessage</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadMessage} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationMessageDeliveryPayloadMessage} The populated <code>ConversationMessageDeliveryPayloadMessage</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationMessageDeliveryPayloadMessage();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ConversationMessageDeliveryPayloadMessage;
}();
/**
 * A string representing the ID of the message.
 * @member {String} id
 */


ConversationMessageDeliveryPayloadMessage.prototype['id'] = undefined;
var _default = ConversationMessageDeliveryPayloadMessage;
exports["default"] = _default;
