// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ConversationLeaveEventAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ConversationLeaveEventAllOfPayload = _interopRequireDefault(require("./ConversationLeaveEventAllOfPayload"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ConversationLeaveEventAllOf model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ConversationLeaveEventAllOf
 * @version 9.4.5
 */
var ConversationLeaveEventAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConversationLeaveEventAllOf</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationLeaveEventAllOf
   */
  function ConversationLeaveEventAllOf() {
    _classCallCheck(this, ConversationLeaveEventAllOf);

    ConversationLeaveEventAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ConversationLeaveEventAllOf, [{
    key: "getPayload",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationLeaveEventAllOfPayload}
         */
    function getPayload() {
      return this.payload;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationLeaveEventAllOfPayload} payload
     */

  }, {
    key: "setPayload",
    value: function setPayload(payload) {
      this['payload'] = payload;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ConversationLeaveEventAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationLeaveEventAllOf} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationLeaveEventAllOf} The populated <code>ConversationLeaveEventAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConversationLeaveEventAllOf();

        if (data.hasOwnProperty('payload')) {
          obj['payload'] = _ConversationLeaveEventAllOfPayload["default"].constructFromObject(data['payload']);
        }
      }

      return obj;
    }
  }]);

  return ConversationLeaveEventAllOf;
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ConversationLeaveEventAllOfPayload} payload
 */


ConversationLeaveEventAllOf.prototype['payload'] = undefined;
var _default = ConversationLeaveEventAllOf;
exports["default"] = _default;
