// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListFilter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The IntegrationListFilter model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListFilter
 * @version 9.4.5
 */
var IntegrationListFilter = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IntegrationListFilter</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListFilter
   */
  function IntegrationListFilter() {
    _classCallCheck(this, IntegrationListFilter);

    IntegrationListFilter.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(IntegrationListFilter, [{
    key: "getTypes",
    value:
    /**
         * Returns Comma-separated list of types to return. If omitted, all types are returned.
         * @return {String}
         */
    function getTypes() {
      return this.types;
    }
    /**
     * Sets Comma-separated list of types to return. If omitted, all types are returned.
     * @param {String} types Comma-separated list of types to return. If omitted, all types are returned.
     */

  }, {
    key: "setTypes",
    value: function setTypes(types) {
      this['types'] = types;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>IntegrationListFilter</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListFilter} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationListFilter} The populated <code>IntegrationListFilter</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IntegrationListFilter();

        if (data.hasOwnProperty('types')) {
          obj['types'] = _ApiClient["default"].convertToType(data['types'], 'String');
        }
      }

      return obj;
    }
  }]);

  return IntegrationListFilter;
}();
/**
 * Comma-separated list of types to return. If omitted, all types are returned.
 * @member {String} types
 */


IntegrationListFilter.prototype['types'] = undefined;
var _default = IntegrationListFilter;
exports["default"] = _default;
