// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/IntegrationResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Integration = _interopRequireDefault(require("./Integration"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The IntegrationResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/IntegrationResponse
 * @version 9.4.5
 */
var IntegrationResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IntegrationResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationResponse
   */
  function IntegrationResponse() {
    _classCallCheck(this, IntegrationResponse);

    IntegrationResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(IntegrationResponse, [{
    key: "getIntegration",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Integration}
         */
    function getIntegration() {
      return this.integration;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Integration} integration
     */

  }, {
    key: "setIntegration",
    value: function setIntegration(integration) {
      this['integration'] = integration;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>IntegrationResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationResponse} The populated <code>IntegrationResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IntegrationResponse();

        if (data.hasOwnProperty('integration')) {
          obj['integration'] = _Integration["default"].constructFromObject(data['integration']);
        }
      }

      return obj;
    }
  }]);

  return IntegrationResponse;
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Integration} integration
 */


IntegrationResponse.prototype['integration'] = undefined;
var _default = IntegrationResponse;
exports["default"] = _default;
