// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/IntegrationApiKeyListResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ApiKey = _interopRequireDefault(require("./ApiKey"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The IntegrationApiKeyListResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/IntegrationApiKeyListResponse
 * @version 9.4.5
 */
var IntegrationApiKeyListResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IntegrationApiKeyListResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationApiKeyListResponse
   */
  function IntegrationApiKeyListResponse() {
    _classCallCheck(this, IntegrationApiKeyListResponse);

    IntegrationApiKeyListResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(IntegrationApiKeyListResponse, [{
    key: "getKeys",
    value:
    /**
         * Returns Integration keys of the supplied integration.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/ApiKey>}
         */
    function getKeys() {
      return this.keys;
    }
    /**
     * Sets Integration keys of the supplied integration.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/ApiKey>} keys Integration keys of the supplied integration.
     */

  }, {
    key: "setKeys",
    value: function setKeys(keys) {
      this['keys'] = keys;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>IntegrationApiKeyListResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationApiKeyListResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/IntegrationApiKeyListResponse} The populated <code>IntegrationApiKeyListResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IntegrationApiKeyListResponse();

        if (data.hasOwnProperty('keys')) {
          obj['keys'] = _ApiClient["default"].convertToType(data['keys'], [_ApiKey["default"]]);
        }
      }

      return obj;
    }
  }]);

  return IntegrationApiKeyListResponse;
}();
/**
 * Integration keys of the supplied integration.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/ApiKey>} keys
 */


IntegrationApiKeyListResponse.prototype['keys'] = undefined;
var _default = IntegrationApiKeyListResponse;
exports["default"] = _default;
