// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageLocation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The LocationMessageLocation model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageLocation
 * @version 9.4.5
 */
var LocationMessageLocation = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>LocationMessageLocation</code>.
   * Information about the location.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageLocation
   */
  function LocationMessageLocation() {
    _classCallCheck(this, LocationMessageLocation);

    LocationMessageLocation.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(LocationMessageLocation, [{
    key: "getAddress",
    value:
    /**
         * Returns A string representing the address of the location.
         * @return {String}
         */
    function getAddress() {
      return this.address;
    }
    /**
     * Sets A string representing the address of the location.
     * @param {String} address A string representing the address of the location.
     */

  }, {
    key: "setAddress",
    value: function setAddress(address) {
      this['address'] = address;
    }
    /**
         * Returns A string representing the name of the location.
         * @return {String}
         */

  }, {
    key: "getName",
    value: function getName() {
      return this.name;
    }
    /**
     * Sets A string representing the name of the location.
     * @param {String} name A string representing the name of the location.
     */

  }, {
    key: "setName",
    value: function setName(name) {
      this['name'] = name;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>LocationMessageLocation</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageLocation} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/LocationMessageLocation} The populated <code>LocationMessageLocation</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new LocationMessageLocation();

        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return LocationMessageLocation;
}();
/**
 * A string representing the address of the location.
 * @member {String} address
 */


LocationMessageLocation.prototype['address'] = undefined;
/**
 * A string representing the name of the location.
 * @member {String} name
 */

LocationMessageLocation.prototype['name'] = undefined;
var _default = LocationMessageLocation;
exports["default"] = _default;
