// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/MessageListResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Links = _interopRequireDefault(require("./Links"));

var _Message = _interopRequireDefault(require("./Message"));

var _Meta = _interopRequireDefault(require("./Meta"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MessageListResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/MessageListResponse
 * @version 9.4.5
 */
var MessageListResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MessageListResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/MessageListResponse
   */
  function MessageListResponse() {
    _classCallCheck(this, MessageListResponse);

    MessageListResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MessageListResponse, [{
    key: "getMessages",
    value:
    /**
         * Returns List of returned messages.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Message>}
         */
    function getMessages() {
      return this.messages;
    }
    /**
     * Sets List of returned messages.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Message>} messages List of returned messages.
     */

  }, {
    key: "setMessages",
    value: function setMessages(messages) {
      this['messages'] = messages;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta}
         */

  }, {
    key: "getMeta",
    value: function getMeta() {
      return this.meta;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta} meta
     */

  }, {
    key: "setMeta",
    value: function setMeta(meta) {
      this['meta'] = meta;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Links}
         */

  }, {
    key: "getLinks",
    value: function getLinks() {
      return this.links;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} links
     */

  }, {
    key: "setLinks",
    value: function setLinks(links) {
      this['links'] = links;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MessageListResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageListResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageListResponse} The populated <code>MessageListResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MessageListResponse();

        if (data.hasOwnProperty('messages')) {
          obj['messages'] = _ApiClient["default"].convertToType(data['messages'], [_Message["default"]]);
        }

        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _Meta["default"].constructFromObject(data['meta']);
        }

        if (data.hasOwnProperty('links')) {
          obj['links'] = _Links["default"].constructFromObject(data['links']);
        }
      }

      return obj;
    }
  }]);

  return MessageListResponse;
}();
/**
 * List of returned messages.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/Message>} messages
 */


MessageListResponse.prototype['messages'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Meta} meta
 */

MessageListResponse.prototype['meta'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Links} links
 */

MessageListResponse.prototype['links'] = undefined;
var _default = MessageListResponse;
exports["default"] = _default;
