// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideApple.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _MessageOverridePayload = _interopRequireDefault(require("./MessageOverridePayload"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The MessageOverrideApple model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideApple
 * @version 9.4.5
 */
var MessageOverrideApple = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MessageOverrideApple</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideApple
   */
  function MessageOverrideApple() {
    _classCallCheck(this, MessageOverrideApple);

    MessageOverrideApple.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(MessageOverrideApple, [{
    key: "getApple",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload}
         */
    function getApple() {
      return this.apple;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} apple
     */

  }, {
    key: "setApple",
    value: function setApple(apple) {
      this['apple'] = apple;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>MessageOverrideApple</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideApple} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverrideApple} The populated <code>MessageOverrideApple</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MessageOverrideApple();

        if (data.hasOwnProperty('apple')) {
          obj['apple'] = _MessageOverridePayload["default"].constructFromObject(data['apple']);
        }
      }

      return obj;
    }
  }]);

  return MessageOverrideApple;
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/MessageOverridePayload} apple
 */


MessageOverrideApple.prototype['apple'] = undefined;
var _default = MessageOverrideApple;
exports["default"] = _default;
