// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserExternalId.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ParticipantLeaveBodyUserExternalId model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserExternalId
 * @version 9.4.5
 */
var ParticipantLeaveBodyUserExternalId = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ParticipantLeaveBodyUserExternalId</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserExternalId
   */
  function ParticipantLeaveBodyUserExternalId() {
    _classCallCheck(this, ParticipantLeaveBodyUserExternalId);

    ParticipantLeaveBodyUserExternalId.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ParticipantLeaveBodyUserExternalId, [{
    key: "getUserExternalId",
    value:
    /**
         * Returns The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
         * @return {String}
         */
    function getUserExternalId() {
      return this.userExternalId;
    }
    /**
     * Sets The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     * @param {String} userExternalId The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     */

  }, {
    key: "setUserExternalId",
    value: function setUserExternalId(userExternalId) {
      this['userExternalId'] = userExternalId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ParticipantLeaveBodyUserExternalId</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserExternalId} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserExternalId} The populated <code>ParticipantLeaveBodyUserExternalId</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ParticipantLeaveBodyUserExternalId();

        if (data.hasOwnProperty('userExternalId')) {
          obj['userExternalId'] = _ApiClient["default"].convertToType(data['userExternalId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ParticipantLeaveBodyUserExternalId;
}();
/**
 * The externalId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} userExternalId
 */


ParticipantLeaveBodyUserExternalId.prototype['userExternalId'] = undefined;
var _default = ParticipantLeaveBodyUserExternalId;
exports["default"] = _default;
