// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyParticipantId.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ParticipantLeaveBodyParticipantId model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyParticipantId
 * @version 9.4.5
 */
var ParticipantLeaveBodyParticipantId = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ParticipantLeaveBodyParticipantId</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyParticipantId
   */
  function ParticipantLeaveBodyParticipantId() {
    _classCallCheck(this, ParticipantLeaveBodyParticipantId);

    ParticipantLeaveBodyParticipantId.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ParticipantLeaveBodyParticipantId, [{
    key: "getParticipantId",
    value:
    /**
         * Returns The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
         * @return {String}
         */
    function getParticipantId() {
      return this.participantId;
    }
    /**
     * Sets The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     * @param {String} participantId The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     */

  }, {
    key: "setParticipantId",
    value: function setParticipantId(participantId) {
      this['participantId'] = participantId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ParticipantLeaveBodyParticipantId</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyParticipantId} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyParticipantId} The populated <code>ParticipantLeaveBodyParticipantId</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ParticipantLeaveBodyParticipantId();

        if (data.hasOwnProperty('participantId')) {
          obj['participantId'] = _ApiClient["default"].convertToType(data['participantId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ParticipantLeaveBodyParticipantId;
}();
/**
 * The participantId of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} participantId
 */


ParticipantLeaveBodyParticipantId.prototype['participantId'] = undefined;
var _default = ParticipantLeaveBodyParticipantId;
exports["default"] = _default;
