// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ParticipantResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Participant = _interopRequireDefault(require("./Participant"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ParticipantResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ParticipantResponse
 * @version 9.4.5
 */
var ParticipantResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ParticipantResponse</code>.
   * The created participant.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantResponse
   */
  function ParticipantResponse() {
    _classCallCheck(this, ParticipantResponse);

    ParticipantResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ParticipantResponse, [{
    key: "getParticipant",
    value:
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Participant}
         */
    function getParticipant() {
      return this.participant;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Participant} participant
     */

  }, {
    key: "setParticipant",
    value: function setParticipant(participant) {
      this['participant'] = participant;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ParticipantResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantResponse} The populated <code>ParticipantResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ParticipantResponse();

        if (data.hasOwnProperty('participant')) {
          obj['participant'] = _Participant["default"].constructFromObject(data['participant']);
        }
      }

      return obj;
    }
  }]);

  return ParticipantResponse;
}();
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Participant} participant
 */


ParticipantResponse.prototype['participant'] = undefined;
var _default = ParticipantResponse;
exports["default"] = _default;
