// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserId.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ParticipantLeaveBodyUserId model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserId
 * @version 9.4.5
 */
var ParticipantLeaveBodyUserId = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ParticipantLeaveBodyUserId</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserId
   */
  function ParticipantLeaveBodyUserId() {
    _classCallCheck(this, ParticipantLeaveBodyUserId);

    ParticipantLeaveBodyUserId.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ParticipantLeaveBodyUserId, [{
    key: "getUserId",
    value:
    /**
         * Returns The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
         * @return {String}
         */
    function getUserId() {
      return this.userId;
    }
    /**
     * Sets The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     * @param {String} userId The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
     */

  }, {
    key: "setUserId",
    value: function setUserId(userId) {
      this['userId'] = userId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ParticipantLeaveBodyUserId</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserId} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ParticipantLeaveBodyUserId} The populated <code>ParticipantLeaveBodyUserId</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ParticipantLeaveBodyUserId();

        if (data.hasOwnProperty('userId')) {
          obj['userId'] = _ApiClient["default"].convertToType(data['userId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ParticipantLeaveBodyUserId;
}();
/**
 * The id of the user that will be removed from the conversation. It will return 404 if the user can’t be found. 
 * @member {String} userId
 */


ParticipantLeaveBodyUserId.prototype['userId'] = undefined;
var _default = ParticipantLeaveBodyUserId;
exports["default"] = _default;
