// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Referral.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ReferralDetails = _interopRequireDefault(require("./ReferralDetails"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Referral model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Referral
 * @version 9.4.5
 */
var Referral = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Referral</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Referral
   */
  function Referral() {
    _classCallCheck(this, Referral);

    Referral.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Referral, [{
    key: "getCode",
    value:
    /**
         * Returns The referral’s identifier.
         * @return {String}
         */
    function getCode() {
      return this.code;
    }
    /**
     * Sets The referral’s identifier.
     * @param {String} code The referral’s identifier.
     */

  }, {
    key: "setCode",
    value: function setCode(code) {
      this['code'] = code;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails}
         */

  }, {
    key: "getDetails",
    value: function getDetails() {
      return this.details;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails} details
     */

  }, {
    key: "setDetails",
    value: function setDetails(details) {
      this['details'] = details;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Referral</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Referral} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Referral} The populated <code>Referral</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Referral();

        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
        }

        if (data.hasOwnProperty('details')) {
          obj['details'] = _ReferralDetails["default"].constructFromObject(data['details']);
        }
      }

      return obj;
    }
  }]);

  return Referral;
}();
/**
 * The referral’s identifier.
 * @member {String} code
 */


Referral.prototype['code'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/ReferralDetails} details
 */

Referral.prototype['details'] = undefined;
var _default = Referral;
exports["default"] = _default;
