// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _SwitchboardIntegration = _interopRequireDefault(require("./SwitchboardIntegration"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SwitchboardIntegrationResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationResponse
 * @version 9.4.5
 */
var SwitchboardIntegrationResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SwitchboardIntegrationResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationResponse
   */
  function SwitchboardIntegrationResponse() {
    _classCallCheck(this, SwitchboardIntegrationResponse);

    SwitchboardIntegrationResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SwitchboardIntegrationResponse, [{
    key: "getSwitchboardIntegration",
    value:
    /**
         * Returns The switchboard integration.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegration}
         */
    function getSwitchboardIntegration() {
      return this.switchboardIntegration;
    }
    /**
     * Sets The switchboard integration.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegration} switchboardIntegration The switchboard integration.
     */

  }, {
    key: "setSwitchboardIntegration",
    value: function setSwitchboardIntegration(switchboardIntegration) {
      this['switchboardIntegration'] = switchboardIntegration;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SwitchboardIntegrationResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationResponse} The populated <code>SwitchboardIntegrationResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SwitchboardIntegrationResponse();

        if (data.hasOwnProperty('switchboardIntegration')) {
          obj['switchboardIntegration'] = _ApiClient["default"].convertToType(data['switchboardIntegration'], _SwitchboardIntegration["default"]);
        }
      }

      return obj;
    }
  }]);

  return SwitchboardIntegrationResponse;
}();
/**
 * The switchboard integration.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegration} switchboardIntegration
 */


SwitchboardIntegrationResponse.prototype['switchboardIntegration'] = undefined;
var _default = SwitchboardIntegrationResponse;
exports["default"] = _default;
