// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationListResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _SwitchboardIntegration = _interopRequireDefault(require("./SwitchboardIntegration"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The SwitchboardIntegrationListResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationListResponse
 * @version 9.4.5
 */
var SwitchboardIntegrationListResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SwitchboardIntegrationListResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationListResponse
   */
  function SwitchboardIntegrationListResponse() {
    _classCallCheck(this, SwitchboardIntegrationListResponse);

    SwitchboardIntegrationListResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(SwitchboardIntegrationListResponse, [{
    key: "getSwitchboardIntegrations",
    value:
    /**
         * Returns List of returned switchboard integrations.
         * @return {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegration>}
         */
    function getSwitchboardIntegrations() {
      return this.switchboardIntegrations;
    }
    /**
     * Sets List of returned switchboard integrations.
     * @param {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegration>} switchboardIntegrations List of returned switchboard integrations.
     */

  }, {
    key: "setSwitchboardIntegrations",
    value: function setSwitchboardIntegrations(switchboardIntegrations) {
      this['switchboardIntegrations'] = switchboardIntegrations;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>SwitchboardIntegrationListResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationListResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegrationListResponse} The populated <code>SwitchboardIntegrationListResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SwitchboardIntegrationListResponse();

        if (data.hasOwnProperty('switchboardIntegrations')) {
          obj['switchboardIntegrations'] = _ApiClient["default"].convertToType(data['switchboardIntegrations'], [_SwitchboardIntegration["default"]]);
        }
      }

      return obj;
    }
  }]);

  return SwitchboardIntegrationListResponse;
}();
/**
 * List of returned switchboard integrations.
 * @member {Array.<module:sunshine-conversations-client/sunshine-conversations-client.model/SwitchboardIntegration>} switchboardIntegrations
 */


SwitchboardIntegrationListResponse.prototype['switchboardIntegrations'] = undefined;
var _default = SwitchboardIntegrationListResponse;
exports["default"] = _default;
