// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/Target.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Target model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/Target
 * @version 9.4.5
 */
var Target = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Target</code>.
   * The target conversation to attach the client to.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/Target
   * @param conversationId {String} The conversation ID of the target conversation.
   */
  function Target(conversationId) {
    _classCallCheck(this, Target);

    Target.initialize(this, conversationId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Target, [{
    key: "getConversationId",
    value:
    /**
         * Returns The conversation ID of the target conversation.
         * @return {String}
         */
    function getConversationId() {
      return this.conversationId;
    }
    /**
     * Sets The conversation ID of the target conversation.
     * @param {String} conversationId The conversation ID of the target conversation.
     */

  }, {
    key: "setConversationId",
    value: function setConversationId(conversationId) {
      this['conversationId'] = conversationId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj, conversationId) {
      obj['conversationId'] = conversationId;
    }
    /**
     * Constructs a <code>Target</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Target} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Target} The populated <code>Target</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Target();

        if (data.hasOwnProperty('conversationId')) {
          obj['conversationId'] = _ApiClient["default"].convertToType(data['conversationId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return Target;
}();
/**
 * The conversation ID of the target conversation.
 * @member {String} conversationId
 */


Target.prototype['conversationId'] = undefined;
var _default = Target;
exports["default"] = _default;
