// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UserResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _User = _interopRequireDefault(require("./User"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UserResponse model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UserResponse
 * @version 9.4.5
 */
var UserResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserResponse</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UserResponse
   */
  function UserResponse() {
    _classCallCheck(this, UserResponse);

    UserResponse.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UserResponse, [{
    key: "getUser",
    value:
    /**
         * Returns The user.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/User}
         */
    function getUser() {
      return this.user;
    }
    /**
     * Sets The user.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/User} user The user.
     */

  }, {
    key: "setUser",
    value: function setUser(user) {
      this['user'] = user;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserResponse} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserResponse} The populated <code>UserResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserResponse();

        if (data.hasOwnProperty('user')) {
          obj['user'] = _ApiClient["default"].convertToType(data['user'], _User["default"]);
        }
      }

      return obj;
    }
  }]);

  return UserResponse;
}();
/**
 * The user.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/User} user
 */


UserResponse.prototype['user'] = undefined;
var _default = UserResponse;
exports["default"] = _default;
