// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UserTruncated model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated
 * @version 9.4.5
 */
var UserTruncated = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserTruncated</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated
   */
  function UserTruncated() {
    _classCallCheck(this, UserTruncated);

    UserTruncated.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UserTruncated, [{
    key: "getId",
    value:
    /**
         * Returns The unique ID of the user.
         * @return {String}
         */
    function getId() {
      return this.id;
    }
    /**
     * Sets The unique ID of the user.
     * @param {String} id The unique ID of the user.
     */

  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
    /**
         * Returns An optional ID that if specified can also be used to retrieve the user. 
         * @return {String}
         */

  }, {
    key: "getExternalId",
    value: function getExternalId() {
      return this.externalId;
    }
    /**
     * Sets An optional ID that if specified can also be used to retrieve the user. 
     * @param {String} externalId An optional ID that if specified can also be used to retrieve the user. 
     */

  }, {
    key: "setExternalId",
    value: function setExternalId(externalId) {
      this['externalId'] = externalId;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserTruncated</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated} The populated <code>UserTruncated</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserTruncated();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('externalId')) {
          obj['externalId'] = _ApiClient["default"].convertToType(data['externalId'], 'String');
        }
      }

      return obj;
    }
  }]);

  return UserTruncated;
}();
/**
 * The unique ID of the user.
 * @member {String} id
 */


UserTruncated.prototype['id'] = undefined;
/**
 * An optional ID that if specified can also be used to retrieve the user. 
 * @member {String} externalId
 */

UserTruncated.prototype['externalId'] = undefined;
var _default = UserTruncated;
exports["default"] = _default;
