// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UserAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Profile = _interopRequireDefault(require("./Profile"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UserAllOf model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UserAllOf
 * @version 9.4.5
 */
var UserAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserAllOf</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UserAllOf
   */
  function UserAllOf() {
    _classCallCheck(this, UserAllOf);

    UserAllOf.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UserAllOf, [{
    key: "getSignedUpAt",
    value:
    /**
         * @return {String}
         */
    function getSignedUpAt() {
      return this.signedUpAt;
    }
    /**
     * @param {String} signedUpAt
     */

  }, {
    key: "setSignedUpAt",
    value: function setSignedUpAt(signedUpAt) {
      this['signedUpAt'] = signedUpAt;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile}
         */

  }, {
    key: "getProfile",
    value: function getProfile() {
      return this.profile;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile} profile
     */

  }, {
    key: "setProfile",
    value: function setProfile(profile) {
      this['profile'] = profile;
    }
    /**
         * @return {Object}
         */

  }, {
    key: "getMetadata",
    value: function getMetadata() {
      return this.metadata;
    }
    /**
     * @param {Object} metadata
     */

  }, {
    key: "setMetadata",
    value: function setMetadata(metadata) {
      this['metadata'] = metadata;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserAllOf} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserAllOf} The populated <code>UserAllOf</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserAllOf();

        if (data.hasOwnProperty('signedUpAt')) {
          obj['signedUpAt'] = _ApiClient["default"].convertToType(data['signedUpAt'], 'String');
        }

        if (data.hasOwnProperty('profile')) {
          obj['profile'] = _ApiClient["default"].convertToType(data['profile'], _Profile["default"]);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
      }

      return obj;
    }
  }]);

  return UserAllOf;
}();
/**
 * @member {String} signedUpAt
 */


UserAllOf.prototype['signedUpAt'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile} profile
 */

UserAllOf.prototype['profile'] = undefined;
/**
 * @member {Object} metadata
 */

UserAllOf.prototype['metadata'] = undefined;
var _default = UserAllOf;
exports["default"] = _default;
