// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/User.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Profile = _interopRequireDefault(require("./Profile"));

var _UserAllOf = _interopRequireDefault(require("./UserAllOf"));

var _UserTruncated = _interopRequireDefault(require("./UserTruncated"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The User model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/User
 * @version 9.4.5
 */
var User = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>User</code>.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/User
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/UserTruncated
   * @implements module:sunshine-conversations-client/sunshine-conversations-client.model/UserAllOf
   */
  function User() {
    _classCallCheck(this, User);

    _UserTruncated["default"].initialize(this);

    _UserAllOf["default"].initialize(this);

    User.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(User, [{
    key: "getId",
    value:
    /**
         * Returns The unique ID of the user.
         * @return {String}
         */
    function getId() {
      return this.id;
    }
    /**
     * Sets The unique ID of the user.
     * @param {String} id The unique ID of the user.
     */

  }, {
    key: "setId",
    value: function setId(id) {
      this['id'] = id;
    }
    /**
         * Returns An optional ID that if specified can also be used to retrieve the user. 
         * @return {String}
         */

  }, {
    key: "getExternalId",
    value: function getExternalId() {
      return this.externalId;
    }
    /**
     * Sets An optional ID that if specified can also be used to retrieve the user. 
     * @param {String} externalId An optional ID that if specified can also be used to retrieve the user. 
     */

  }, {
    key: "setExternalId",
    value: function setExternalId(externalId) {
      this['externalId'] = externalId;
    }
    /**
         * @return {String}
         */

  }, {
    key: "getSignedUpAt",
    value: function getSignedUpAt() {
      return this.signedUpAt;
    }
    /**
     * @param {String} signedUpAt
     */

  }, {
    key: "setSignedUpAt",
    value: function setSignedUpAt(signedUpAt) {
      this['signedUpAt'] = signedUpAt;
    }
    /**
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile}
         */

  }, {
    key: "getProfile",
    value: function getProfile() {
      return this.profile;
    }
    /**
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile} profile
     */

  }, {
    key: "setProfile",
    value: function setProfile(profile) {
      this['profile'] = profile;
    }
    /**
         * @return {Object}
         */

  }, {
    key: "getMetadata",
    value: function getMetadata() {
      return this.metadata;
    }
    /**
     * @param {Object} metadata
     */

  }, {
    key: "setMetadata",
    value: function setMetadata(metadata) {
      this['metadata'] = metadata;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>User</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/User} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/User} The populated <code>User</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new User();

        _UserTruncated["default"].constructFromObject(data, obj);

        _UserAllOf["default"].constructFromObject(data, obj);

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('externalId')) {
          obj['externalId'] = _ApiClient["default"].convertToType(data['externalId'], 'String');
        }

        if (data.hasOwnProperty('signedUpAt')) {
          obj['signedUpAt'] = _ApiClient["default"].convertToType(data['signedUpAt'], 'String');
        }

        if (data.hasOwnProperty('profile')) {
          obj['profile'] = _ApiClient["default"].convertToType(data['profile'], _Profile["default"]);
        }

        if (data.hasOwnProperty('metadata')) {
          obj['metadata'] = _ApiClient["default"].convertToType(data['metadata'], Object);
        }
      }

      return obj;
    }
  }]);

  return User;
}();
/**
 * The unique ID of the user.
 * @member {String} id
 */


User.prototype['id'] = undefined;
/**
 * An optional ID that if specified can also be used to retrieve the user. 
 * @member {String} externalId
 */

User.prototype['externalId'] = undefined;
/**
 * @member {String} signedUpAt
 */

User.prototype['signedUpAt'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile} profile
 */

User.prototype['profile'] = undefined;
/**
 * @member {Object} metadata
 */

User.prototype['metadata'] = undefined; // Implement UserTruncated interface:

/**
 * The unique ID of the user.
 * @member {String} id
 */

_UserTruncated["default"].prototype['id'] = undefined;
/**
 * An optional ID that if specified can also be used to retrieve the user. 
 * @member {String} externalId
 */

_UserTruncated["default"].prototype['externalId'] = undefined; // Implement UserAllOf interface:

/**
 * @member {String} signedUpAt
 */

_UserAllOf["default"].prototype['signedUpAt'] = undefined;
/**
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/Profile} profile
 */

_UserAllOf["default"].prototype['profile'] = undefined;
/**
 * @member {Object} metadata
 */

_UserAllOf["default"].prototype['metadata'] = undefined;
var _default = User;
exports["default"] = _default;
