// sunshine-conversations-client/dist/sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedUsers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _User = _interopRequireDefault(require("./User"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The UserMergeEventAllOfPayloadMergedUsers model module.
 * @module sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedUsers
 * @version 9.4.5
 */
var UserMergeEventAllOfPayloadMergedUsers = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserMergeEventAllOfPayloadMergedUsers</code>.
   * Contains information about the users that were merged together.
   * @alias module:sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedUsers
   */
  function UserMergeEventAllOfPayloadMergedUsers() {
    _classCallCheck(this, UserMergeEventAllOfPayloadMergedUsers);

    UserMergeEventAllOfPayloadMergedUsers.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(UserMergeEventAllOfPayloadMergedUsers, [{
    key: "getSurviving",
    value:
    /**
         * Returns The user that now represents the merged user object.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/User}
         */
    function getSurviving() {
      return this.surviving;
    }
    /**
     * Sets The user that now represents the merged user object.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/User} surviving The user that now represents the merged user object.
     */

  }, {
    key: "setSurviving",
    value: function setSurviving(surviving) {
      this['surviving'] = surviving;
    }
    /**
         * Returns The user that was unified into the surviving user object.
         * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/User}
         */

  }, {
    key: "getDiscarded",
    value: function getDiscarded() {
      return this.discarded;
    }
    /**
     * Sets The user that was unified into the surviving user object.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/User} discarded The user that was unified into the surviving user object.
     */

  }, {
    key: "setDiscarded",
    value: function setDiscarded(discarded) {
      this['discarded'] = discarded;
    }
  }], [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>UserMergeEventAllOfPayloadMergedUsers</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedUsers} obj Optional instance to populate.
     * @return {module:sunshine-conversations-client/sunshine-conversations-client.model/UserMergeEventAllOfPayloadMergedUsers} The populated <code>UserMergeEventAllOfPayloadMergedUsers</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserMergeEventAllOfPayloadMergedUsers();

        if (data.hasOwnProperty('surviving')) {
          obj['surviving'] = _ApiClient["default"].convertToType(data['surviving'], _User["default"]);
        }

        if (data.hasOwnProperty('discarded')) {
          obj['discarded'] = _ApiClient["default"].convertToType(data['discarded'], _User["default"]);
        }
      }

      return obj;
    }
  }]);

  return UserMergeEventAllOfPayloadMergedUsers;
}();
/**
 * The user that now represents the merged user object.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/User} surviving
 */


UserMergeEventAllOfPayloadMergedUsers.prototype['surviving'] = undefined;
/**
 * The user that was unified into the surviving user object.
 * @member {module:sunshine-conversations-client/sunshine-conversations-client.model/User} discarded
 */

UserMergeEventAllOfPayloadMergedUsers.prototype['discarded'] = undefined;
var _default = UserMergeEventAllOfPayloadMergedUsers;
exports["default"] = _default;
